src/solvers/prop/solver_resource_limits.h

```diff
@@ -16,7 +16,7 @@ class solver_resource_limitst
 {
 public:
   /// Set the limit for the solver to time out in seconds
-  virtual void set_time_limit_seconds(uint32_t) = 0;
+  virtual bool set_time_limit_seconds(uint32_t) = 0;
```

Comment:
return value should be documented at least, or better yet we could use a status enum to indicate whether this succeeded or failed (because we use `bool` for success/failure inconsistently throughout the code base, but most of the time `true` means _failure_ which is the opposite to how it works here).

Also, should the `return false;` implementation here be the default?